core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1146,16 +1146,18 @@ public RelBuilder functionScan(SqlOperator operator,
    * and optimized in a similar way to the {@link #and} method.
    * If the result is TRUE no filter is created. */
   public RelBuilder filter(RexNode... predicates) {
-    return filter(ImmutableList.copyOf(predicates));
+    return filter(Collections.emptyList(), ImmutableList.copyOf(predicates));
   }
 
-  /** Creates a {@link Filter} of a list of
-   * predicates.
+  /**
+   * Creates a {@link Filter} of a list of
+   * predicates and correlation variables
    *
    * <p>The predicates are combined using AND,
    * and optimized in a similar way to the {@link #and} method.
    * If the result is TRUE no filter is created. */
-  public RelBuilder filter(Iterable<? extends RexNode> predicates) {
+  public RelBuilder filter(Iterable<CorrelationId> correlVariables,
+      Iterable<? extends RexNode> predicates) {
```

Comment:
Can we always keep the `correlVariables` as the last argument ? Just like the `FilterFactory`.

Reply:
Sure. Done